### PR TITLE
doc: cut off v0.4.0 release

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
doc: cut off v0.4.0 release

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
doc: cut off v0.4.0 release
```

### CODE REVIEW

File: charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["csi.storage.k8s.io"]
+    resources: ["csinodeinfos"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["volumeattachments"]
+    verbs: ["get", "list", "watch", "update"]
+
```
Comment (ZeroMagic) [inline charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml]: add missing lease RBAC

File: charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents"]
+    verbs: ["create", "get", "list", "watch", "update", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshots"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["create", "list", "watch", "delete"]
+
```
Comment (ZeroMagic) [inline charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml]: add missing lease RBAC. The same problem is also in the YAML file under the deploy directory.

File: charts/v0.4.0/azurefile-csi-driver/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: v0.4.0
+description: Azure file Container Storage Interface (CSI) Storage Plugin
```
Comment (ZeroMagic) [inline charts/v0.4.0/azurefile-csi-driver/Chart.yaml]: file needs to be capitalized

File: charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml (hunk 2 above)
Comment (andyzhangx) [reply charts/v0.4.0/azurefile-csi-driver/templates/rbac-csi-azurefile-controller.yaml]: good catch, added in another commit

**Pull request merged**